includes/class-events.php

```diff
@@ -23,15 +26,37 @@ protected function class_init() {
 		// Prime lock cache if not present
 		Lock::prime_lock( self::LOCK );
 
-		// Prepare environment as early as possible
+		// Prime options and prepare environment as early as possible
 		$earliest_action = did_action( 'muplugins_loaded' ) ? 'plugins_loaded' : 'muplugins_loaded';
+		add_action( $earliest_action, array( $this, 'prime_options' ) );
 		add_action( $earliest_action, array( $this, 'prepare_environment' ) );
 
 		// Allow code loaded as late as the theme to modify the whitelist
 		add_action( 'after_setup_theme', array( $this, 'populate_concurrent_action_whitelist' ) );
 	}
 
 	/**
+	 * Set initial options that control event behaviour
+	 */
+	public function prime_options() {
+		// Prime option to disable automatic execution
+		if ( is_admin() || ( defined( 'WP_CLI' ) && \WP_CLI ) || false !== get_endpoint_type() ) {
+			add_option( self::DISABLE_RUN_OPTION, 0, null, false );
```

Comment:
Given the limited applicability, I'm inclined to remove the priming entirely, and move the `get_option()` call to `run_disabled()`. The option is only checked during CLI and REST requests, so loading it on every request is unnecessary.